### PR TITLE
Make header visible by using redux store

### PR DESCRIPTION
**Pull request by tkhoma**
Here I created new reduce function header and added new visibleHeader variable to the store.
Upgraded react to the latest version

### CODE REVIEW

File: src/components/Present.js [hunk 1]
```diff
@@ -11,11 +13,16 @@ export default class Present extends Component {
...
+            this.props.dispatch({
+                type: types.TOGGLE_HEADER,
+                visibleHeader: true
+            });
         } else {
-            navbar.classList.remove('visible-header');
+            this.props.dispatch({
+                type: types.TOGGLE_HEADER,
+                visibleHeader: false
+            });
```
Comment (aivanochko) [inline src/components/Present.js]: Please use shorthand if/else

```
this.props.dispatch({ 
  type: types.TOGGLE_HEADER,
  visibleHeader: (window.scrollY > viewportHeight - 100) ? true : false
});
```

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -49,4 +49,4 @@ Footer.propTypes = {};
 
 Footer.defaultProps = {};
```
Comment (oziniak) [inline src/components/Footer/Footer.js]: redundant. either add proptypes of remove this.

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -1,44 +1,38 @@
 import React, { Component } from 'react';
 import { NavLink } from 'react-router-dom'
 import Menu from './Menu';
+import { connect } from 'react-redux';
+import * as types from '../../ActionTypes';
```
Comment (oziniak) [inline src/components/Header/Header.js]: it's better not to use relative paths

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -1,44 +1,38 @@
...
-    
-    // TODO: need to use redux state instead of this
+
     componentDidMount() {
         if (document.location.pathname !== '/') {
-            let navbar = document.querySelector('nav.header');
-            navbar.classList.add('visible-header');
+            this.props.dispatch({
+                type: types.TOGGLE_HEADER,
+                visibleHeader: true
```
Comment (oziniak) [inline src/components/Header/Header.js]: convention for action is a shape of `{type: ..., payload: ...}`

File: src/components/Header/Header.js [hunk 5]
```diff
@@ -1,44 +1,38 @@
...
     }
-    
-    componentWillReceiveProps(newProps) {
-        this.setState({
-            user: newProps.user
-        });
-    }
-    
+
     handleScroll(event) {
```
Comment (oziniak) [inline src/components/Header/Header.js:23]: it's better to debounce or throttle scroll events

File: src/components/Login/LoginForm.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import * as types from '../../ActionTypes';
+
+class LoginForm extends Component {
+    constructor(props) {
+        super()
+    }
+
+    componentDidMount() {
```
Comment (oziniak) [inline src/components/Login/LoginForm.js]: I see the same logic being reused in `componentDidMount`. you might want to try implementing it in HOC.

File: src/redux/reducers/header.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import * as types from '../../ActionTypes';
+
+export default function headerVisibility(state = [], action) {
+    switch(action.type) {
+        case types.TOGGLE_HEADER:
+            if (state.visibleHeader == action.visibleHeader) {
```
Comment (oziniak) [inline src/redux/reducers/header.js:6]: better dispatching this action only when the change is needed

**Review by oziniak — APPROVED** [review]

**Review by oziniak — APPROVED** [review]
test

**Pull request merged**